### PR TITLE
Add initial ML blog post

### PR DESCRIPTION
**Pull request by ericholscher**
This is meant to be "high level" and talk about the problems & benefits,
but I'm not sure if I really achieved that.

I'm also not in love with my statement of the problem.
I think there's definitely a cleaner way to communicate it,
and clarify it a bit more.

**Feedback**: This is a rough draft, so looking for feedback on high level concepts & structure.

### CODE REVIEW

File: content/posts/2022-content-based-targeting.md [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+description: Content-based targeting allows us to provide great ad targeting without the privacy downsides of user-based tracking
+tags: content-targeting, advertisers
+authors: Eric Holscher 
+image: /images/posts/2022-bullseye.jpg
+image_credit: <span>Photo by <a href="https://unsplash.com/@silvanarnet?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Silvan Arnet</a> on <a href="https://unsplash.com/s/photos/bullseye?utm_source=unsplash&utm_medium=referral&utm_content=creditCopyText">Unsplash</a></span>
+
+
+The advertising industry was born with the idea of content-based targeting.
+You had a message you wanted to send,
+and you picked the audience based on where you could reach them.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:12]:
```suggestion
and you picked the right channel to reach your target audience.
```

File: content/posts/2022-content-based-targeting.md [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+Your ad space was to stand in a food market,
+a specific newspaper section,
+or which city block for a poster.
+
+Recently the industry has gained the technology to track and target users,
+which [made it irrelevant where an ad was shown](https://www.ethicalads.io/blog/2021/05/how-invasive-ad-targeting-is-bad-for-journalism-and-other-high-quality-publishers/).
+Unfortunately this also lead to massive troves of data being collected about all users,
+which has many unintended consequences.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:20]: and has left users squeamish?

File: content/posts/2022-content-based-targeting.md [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+
+## TODO: Remove first 2 paragraphs?
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:22]: We have gone 2 paragraphs without mentioning any ML. Gotta get to that lede faster.

File: content/posts/2022-content-based-targeting.md [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+In the past few years,
+advancements in machine learning have made a new era of content-based ad targeting possible.
+Computers are now stunningly good at learning how to categorize documents,
+which allows us to automate and scale content-based targeting to our entire network.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:27]: I think this paragraph is good to lead the post with. Maybe have 1 sentence about the downsides of invasive targeting (from the above 2 paragraphs you're cutting) in here.

File: content/posts/2022-content-based-targeting.md [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+
+This post will cover how our content-based targeting works at a high level,
+and the implications for the future growth of our business.
+We launched EthicalAds with the vision that user-based tracking was bad for privacy,
+and we're working to build a technical solution to that problem.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:32]: I'd just mention that we're aiming to build a technical solution that still targets fantastically but without the downsides.

File: content/posts/2022-content-based-targeting.md [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+
+Building an ad network requires us to serve ads across a number of different sites with differing content.
+Many of these sites as well have content that focuses on a number of different topics.
+You can imagine a blog with both Python and Kubernetes content,
+and the audience for each of those posts is different.
+**Each page of content that we serve should have targeting based on what it's about.**
+
+The other major problem is trying to find ways to bundle content together,
+so that advertisers can buy a large amount of it.
+If we can only offer $500 a month of a very specific audience,
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:44]: I'd cut this down to $100.

File: content/posts/2022-content-based-targeting.md [hunk 7]
```diff
@@ -0,0 +1,100 @@
...
+and have reasonably good targeting.
+It had a few drawbacks though:
+
+* It needed users to tag content, and most pages aren't tagged (eg. software documentation)
+* We needed to combine tags into topics, and there will always be some tags we don't know about
+* The automatic tagging was quite naive, and had some false positives
+
+We wanted to build a more robust solution that was able to understand a lot more about each page.
+
+### The solution: top down using machine learning
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:67]: I don't know if bottom up vs. top down really explains this well. If anything, I'd call ML bottom up...

File: content/posts/2022-content-based-targeting.md [hunk 8]
```diff
@@ -0,0 +1,100 @@
...
+
+Our approach to machine learning (ML) is more of a top down approach.
+It works by having us train a model what each of our topics are,
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:70]: We're missing an explanation of these topics are the groups of keywords that advertisers frequently buy.

File: content/posts/2022-content-based-targeting.md [hunk 9]
```diff
@@ -0,0 +1,100 @@
...
+by learning from an initial set of human categorized data.
+This allows us to focus on the topics we have enough traffic to effectively target,
+and then teach the model about any new topics quickly as we expand.
+
+When we "train" the ML model,
+it is creating a huge category words and how they map to a topic.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:76]: This sentence doesn't make sense. I'm not sure precisely what it's supposed to say.

File: content/posts/2022-content-based-targeting.md [hunk 10]
```diff
@@ -0,0 +1,100 @@
...
+* We can easily add more topics, as long as we can categorize a small number of pages of content that belong to that topic.
+
+We're excited to expand our ML targeting in the future.
+
+### Next steps: more topics and more privacy
+
+The ease of adding new topics to our models allows us to continue to expand the topics we're able to target.
+In the future,
+if an advertiser comes to us wanting to target a specific subset of content,
+we should be able to train a deploy this targeting within a week.
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:95]: Should we also explain how a bunch of $100 niches become targetable as we automate more and scale up in terms of traffic? Maybe no need to say that.

**Review by davidfischer — COMMENTED** [review]
Overall, it's getting close. I think we need to get right into the ML content. This post does seem a little technical for advertisers. For advertisers, we should be hammering home the message that this will let them target their audience really well without the need for targeting.

File: content/posts/2022-content-based-targeting.md (hunk 7 above)
Comment (ericholscher) [reply content/posts/2022-content-based-targeting.md:51]: I was viewing it as how we start with a topic. For example the old way to build from keywords to topics, but yea, it's not super obvious.

File: content/posts/2022-content-based-targeting.md (hunk 8 above)
Comment (ericholscher) [reply content/posts/2022-content-based-targeting.md:54]: We don't use keywords in the ML approach. That's kinda the whole point right, is that the machine figures out way more keywords than we ever could?

**Review by ericholscher — COMMENTED** [review]
Updated with review 👍

File: content/posts/2022-content-based-targeting.md (hunk 8 above)
Comment (davidfischer) [reply content/posts/2022-content-based-targeting.md:54]: Sure, but the explanation that topics are what advertisers frequently buy is missing.

File: content/posts/2022-content-based-targeting.md [hunk 11]
```diff
@@ -0,0 +1,86 @@
+Title: A new era of content-based targeting for advertising
+Date: April 26, 2022
```
Comment (davidfischer) [inline content/posts/2022-content-based-targeting.md:2]:
```suggestion
Date: November 7, 2022
```

**Review by davidfischer — APPROVED** [review]
I think it's close enough to ship. Don't forget to change the date.

**Pull request merged**